src/os/windows/win32.cpp

```diff
@@ -192,27 +80,15 @@ void FiosGetDrives(FileList &file_list)
 	}
 }
 
-bool FiosIsValidFile(const std::string &, const struct dirent *ent, struct stat *sb)
+bool FiosIsHiddenFile(const std::filesystem::path &path)
 {
-	/* hectonanoseconds between Windows and POSIX epoch */
-	static const int64_t posix_epoch_hns = 0x019DB1DED53E8000LL;
-	const WIN32_FIND_DATA *fd = &ent->dir->fd;
-
-	sb->st_size  = ((uint64_t) fd->nFileSizeHigh << 32) + fd->nFileSizeLow;
-	/* UTC FILETIME to seconds-since-1970 UTC
-	 * we just have to subtract POSIX epoch and scale down to units of seconds.
-	 * http://www.gamedev.net/community/forums/topic.asp?topic_id=294070&whichpage=1&#1860504
-	 * XXX - not entirely correct, since filetimes on FAT aren't UTC but local,
-	 * this won't entirely be correct, but we use the time only for comparison. */
-	sb->st_mtime = (time_t)((*(const uint64_t*)&fd->ftLastWriteTime - posix_epoch_hns) / 1E7);
-	sb->st_mode  = (fd->dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)? S_IFDIR : S_IFREG;
-
-	return true;
-}
+	UINT sem = SetErrorMode(SEM_FAILCRITICALERRORS); // Disable 'no-disk' message box.
 
-bool FiosIsHiddenFile(const struct dirent *ent)
-{
-	return (ent->dir->fd.dwFileAttributes & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)) != 0;
+	DWORD attributes = GetFileAttributes(path.string().c_str());
```

Comment:
```suggestion
	DWORD attributes = GetFileAttributes(OTTD2FS(path.string()).c_str());
```

Reply:
It's already in native format straight from the directory_iterator, so don't think that is right.

Reply:
`GetFileAttributes()` wants an LPCWSTR (`const wchar*`) and `path.string().c_str()` is `const char*`.